warcprox/dedup.py

```diff
@@ -107,9 +112,16 @@ def decorate_with_dedup_info(dedup_db, recorded_url, base32=False):
             and recorded_url.response_recorder.payload_size() > 0):
         digest_key = warcprox.digest_str(recorded_url.response_recorder.payload_digest, base32)
         if recorded_url.warcprox_meta and "captures-bucket" in recorded_url.warcprox_meta:
-            recorded_url.dedup_info = dedup_db.lookup(digest_key, recorded_url.warcprox_meta["captures-bucket"])
+            if isinstance(dedup_db, CdxServerDedup):
+                recorded_url.dedup_info = dedup_db.lookup(digest_key, recorded_url.warcprox_meta["captures-bucket"],
+                                                          recorded_url)
+            else:
+                recorded_url.dedup_info = dedup_db.lookup(digest_key, recorded_url.warcprox_meta["captures-bucket"])
         else:
-            recorded_url.dedup_info = dedup_db.lookup(digest_key)
+            if isinstance(dedup_db, CdxServerDedup):
+                recorded_url.dedup_info = dedup_db.lookup(digest_key, recorded_url)
+            else:
+                recorded_url.dedup_info = dedup_db.lookup(digest_key)
```

Comment:
☝️ This isinstance() stuff is ugly. Please do it a different way. :) Either make all implementations of `lookup()` accept an optional parameter `recorded_url`, or override `decorate_with_dedup_info()` in your subclass, or change the `digest_key` parameter to `key` and pass `recorded_url` as the key, or something...

Reply:
I implemented the first of your suggestions. I didn't want to touch the other dedup classes, this is why I used ``isinstance()``. It is fine now.